File: solutions/9/9.c
#include <math.h>
#include <stdio.h>

int isInteger(double number) {
    return floor(number) == number;
}

int specialPythagoreanTriplet(int triplet_sum) {
    for (int a = 1; a < triplet_sum; a++) {
        for (int b = a + 1; b < triplet_sum - a; b++) {
            double c = sqrt(a * a + b * b);

            if (isInteger(c) && a + b + c == triplet_sum) {
                return a * b * c;
            }
        }
    }

    return -1;
}

int main() {
    int result = specialPythagoreanTriplet(1000);

    printf("%d\n", result);

    return 0;
}

File: solutions/9/9.js
function specialPythagoreanTriplet(triplet_sum) {
    for (let a = 1; a < triplet_sum; a++) {
        for (let b = a + 1; b < triplet_sum - a; b++) {
            const c = Math.sqrt(a * a + b * b);

            if (Number.isInteger(c) && a + b + c === triplet_sum) {
                return a * b * c;
            }
        }
    }

    return -1;
}

function main() {
    const result = specialPythagoreanTriplet(1000);

    console.log(result);
}

main();
